### PR TITLE
Adding models and Test solution

### PR DESCRIPTION
**Pull request by AymericNoel**

### CODE REVIEW

File: src/Trakx.Coinbase.Custody.Client/Models/Transaction.cs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        public string Currency { get; set; }
+
+        [JsonPropertyName("fee")]
+        public long Fee { get; set; }
+
+        [JsonPropertyName("hashes")]
+        public string[] Hashes { get; set; }
+    }
+
+    public class TransactionType
```
[the remark this answers is not in the document]
Comment (AymericNoel) [reply src/Trakx.Coinbase.Custody.Client/Models/Transaction.cs:56]:
```suggestion
    public static class TransactionType
```

File: src/Trakx.Coinbase.Custody.Client/Models/Transaction.cs [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        /// </summary>
+        public static readonly string CoinbaseDeposit = "coinbase_deposit";
+
+        /// <summary>
+        /// Coinbase Custody refund for the leftover amount for a CPFP (child pays for parent) transaction
+        /// </summary>
+        public static readonly string CoinbaseRefund = "coinbase_refund";
+    }
+
+    public class TransactionState
```
[the remark this answers is not in the document]
Comment (AymericNoel) [reply src/Trakx.Coinbase.Custody.Client/Models/Transaction.cs:124]:
```suggestion
    public static class TransactionState
```

File: src/Trakx.Coinbase.Custody.Client/Models/Address.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        public DateTimeOffset CreatedAt { get; set; }
+
+        [JsonPropertyName("updated_at")]
+        public DateTimeOffset UpdatedAt { get; set; }
+
+        [JsonPropertyName("currency")]
+        public string Currency { get; set; }
+    }
+
+    public class AddressState
```
[the remark this answers is not in the document]
Comment (AymericNoel) [reply src/Trakx.Coinbase.Custody.Client/Models/Address.cs:34]:
```suggestion
    public static class AddressState
```

File: src/Trakx.Coinbase.Custody.Client/Models/Transaction.cs [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        public long Fee { get; set; }
+
+        [JsonPropertyName("hashes")]
+        public string[] Hashes { get; set; }
+    }
+
+    public static class TransactionType
+    {
+        /// <summary>
+        /// On-chain deposit of funds to an associated address
```
Comment (monsieurleberre) [inline src/Trakx.Coinbase.Custody.Client/Models/Transaction.cs:59]: Thanks for taking the time to translate the comments in XML doc 🙏

**Pull request merged**